src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -161,8 +161,19 @@ static bool ParsePath(CSharpSyntaxNode? expressionSyntax, bool enabledNullable,
 			}
 			var indexExpression = argumentList[0].Expression;
 			var indexValue = context.SemanticModel.GetConstantValue(indexExpression).Value;
+			if (indexValue is null)
+			{
+				return false;
+			}
+
+			if (!ParsePath(elementAccess.Expression, enabledNullable, context, parts))
+			{
+				return false;
+			}
 
-			return ParsePath(elementAccess.Expression, enabledNullable, context, parts, index: indexValue);
+			var defaultMemberName = "Item"; // TODO we need to check the value of the `[DefaultMemberName]` attribute on the member type
```

Comment:
I presume this will require using `SemanticModel`, right?

Reply:
Yeah, it will be complex and we need to examine the right type... but let's keep that for next week (or later).